src/components/diagrams/diagram-grid-layout.tsx

```diff
@@ -297,7 +300,9 @@ function DiagramGridLayout({ studyUuid, showInSpreadsheet, visible }: Readonly<D
                     filterUuid: diagram.filterUuid,
                     voltageLevelIds: [...diagram.voltageLevelIds, voltageLevelIdToAdd],
                     voltageLevelToExpandIds: diagram.voltageLevelToExpandIds,
-                    voltageLevelToOmitIds: diagram.voltageLevelToOmitIds,
+                    voltageLevelToOmitIds: diagram.voltageLevelToOmitIds.includes(voltageLevelIdToAdd)
```

Comment:
Since includes and filter are both O(n) operation, we can skip the includes and filter immediately